rcsb/workflow/bcif/task_functions.py

```diff
@@ -0,0 +1,370 @@
+)
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+
+def convertPrereleaseCifFiles(
+    listFileName: str,
+    listFileBase: str,
+    remotePath: str,
+    updateBase: str,
+    outfileSuffix: str,
+    contentType: str,
+    outputContentType: bool,
+    outputHash: bool,
+    batch: int,
+    maxFiles: int,
+    maxTempFiles: int,
+    pdbxDict: str,
+    maDict: str,
+    rcsbDict: str,
+) -> bool:
+    """runs once per list file"""
+
+    # paths for randomly-named temp files (bulk removal periodically)
+    temppaths = []
+
+    # read sublist
+    filepath = os.path.join(listFileBase, listFileName)
+    logger.info(
+        "convert prerelease cif files reading list file %s and remote path %s",
+        filepath,
+        remotePath,
+    )
+    files = []
+    if not os.path.exists(filepath):
+        raise FileNotFoundError("no input files")
+    f = open(filepath, "r", encoding="utf-8")
+    for line in f:
+        files.append(line.strip())
+        if 0 < maxFiles <= len(files):
+            break
+    f.close()
+    if len(files) < 1:
+        raise ValueError("no files")
+
+    # determine batch size
+    if (batch is None) or not str(batch).isdigit():
+        batch = 1
+    batch = int(batch)
+    if batch <= 0:
+        batch = multiprocessing.cpu_count()
+    logger.info("distributing %d files across %d sublists", len(files), batch)
+
+    # form dictionary object
+    dictionaryApi = getDictionaryApi(pdbxDict, maDict, rcsbDict)
+
+    # traverse sublist and send each input file to converter
+    procs = []
+    if batch == 1:
+        temppath = tempfile.mkdtemp()
+        temppaths.append(temppath)
+        # process one file at a time
+        for line in files:
+            args = (
+                line,
+                remotePath,
+                updateBase,
+                outfileSuffix,
+                outputContentType,
+                outputHash,
+                contentType,
+                dictionaryApi,
+                temppath,
+                maxTempFiles,
+            )
+            singleTask(*args)
+    else:
+        # process with file batching
+        nfiles = len(files)
+        tasks = splitList(nfiles, batch, files)
+        nresults = len(list(chain(*tasks)))
+        if nresults != nfiles:
+            logger.warning(
+                "split list returned %d files instead of %d", nresults, nfiles
+            )
+        for task in tasks:
+            temppath = tempfile.mkdtemp()
+            temppaths.append(temppath)
+            args = (
+                task,
+                remotePath,
+                updateBase,
+                outfileSuffix,
+                outputContentType,
+                outputHash,
+                contentType,
+                dictionaryApi,
+                temppath,
+                maxTempFiles,
+            )
+            procs.append(multiprocessing.Process(target=batchTask, args=args))
+        for p in procs:
+            p.start()
```

Comment:
Here you seem to be spawning subprocesses with the different subtasks, right? I think it should rather be done all within python, i.e. execute `singleTask()` in parallel. The best way to do it is something like this [SO post](https://stackoverflow.com/a/20549090) recommends.